app/javascript/components/map/components/basemaps/basemaps.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import withTooltipEvt from 'components/ui/with-tooltip-evt';
+import { setMapSettings, setLandsatBasemap } from 'components/map/map-actions';
+import {
+  getBasemap,
+  getLabels,
+  getLayers,
+  getMapZoom,
+  getActiveBoundaries,
+  getBoundaryDatasets
+} from 'components/map/map-selectors';
+import BasemapsComponent from './basemaps-component';
+
+function mapStateToProps(state) {
+  return {
+    layers: getLayers(state.location),
+    mapZoom: getMapZoom(state.location),
+    activeLabels: getLabels(state.location),
+    activeBasemap: getBasemap(state.location),
+    boundaries: getBoundaryDatasets(state.datasets),
+    activeBoundaries: getActiveBoundaries({
+      ...state.datasets,
+      ...state.location
+    })
+  };
+}
+
+class BasemapsContainer extends React.Component {
+  static propTypes = {
+    layers: PropTypes.array,
+    activeBoundaries: PropTypes.object,
+    setMapSettings: PropTypes.func.isRequired,
+    setLandsatBasemap: PropTypes.func.isRequired
+  };
+
+  selectBasemap = (basemap, year) => {
+    if (basemap.dynamic) {
+      if (basemap.id === 'landsat') {
+        return this.props.setLandsatBasemap(year, basemap.defaultUrl);
+      }
+    }
+    return this.props.setMapSettings({ basemap });
+  };
+
+  selectLabels = label => this.props.setMapSettings({ label });
+
+  selectBoundaries = item => {
+    const { layers, activeBoundaries } = this.props;
+    const filteredLayers = activeBoundaries
+      ? layers.filter(l => l.dataset !== activeBoundaries.dataset)
+      : layers;
+    if (item.value) {
+      const newLayers = [
+        {
+          ...item.value,
```

Comment:
We need to make sure we are also passing the dataset. When I pass this it breaks the map for me and I can see in the state there is no dataset defined.

Reply:
this got broken on the merge, can't really fix it by myself need to talk